### PR TITLE
Basic <Video> tag implementation

### PR DESCRIPTION
**Pull request by brentvatne**
Put this together together today with the help of @tadeuzagallo, far from complete but you can see it in action in my [react-native-login repo](https://github.com/brentvatne/react-native-login).

Special props beyond the standard `RCTView`:
- `resizeMode` with the same options as `Image`
- `source` which just accepts a string currently, which is then used to locate the resource locally.

demo gif

Some remaining tasks:
- [ ] Support `require('video!...')`
  - [ ] Support other extensions than just mp4?
- [ ] Add prop to set repeat (none or forever)
- [ ] Restart video when view becomes active again?
- [ ] Decide how far to go with this PR - many more properties and options we could potentially implement

### CODE REVIEW

File: Libraries/Video/VideoContentModes.m [hunk 1]
```diff
@@ -0,0 +1,14 @@
+#import "VideoContentModes.h"
+@import MediaPlayer;
+
+@implementation VideoContentModes
+
+- (NSDictionary *)constantsToExport
```
Comment (tadeuzagallo) [inline Libraries/Video/VideoContentModes.m]: No need for this extra module, you can move it into `RCTVideoManager.m`

**Comment by joshhornby** [comment]
Really like the look of this :+1:

**Comment by grp** [comment]
Hey, this looks great but in the past I've noticed a few issues with `MPMoviePlayerController` — it often blocks the main thread (causing stutters) whenever you make it, and you can only have one of them on the screen at a time. Using `AVPlayer` directly can avoid that, although it's more complex to use.

Internally, we have a wrapper around `AVPlayer` with a nice API. I'll see about open sourcing that: it might make a good core for a `<Video>` component.

**Comment by brentvatne** [comment]
@grp - that would be great! Looking forward to seeing it

**Comment by vjeux** [comment]
Actually, I think this would be better for you to create a separate github repo for this specific component, iterate on it and when it reaches maturity we can see about including it in the main repo. Otherwise we're going to slow you down as we're not going to include something half baked as part of the core build.

**Comment by brentvatne** [comment]
@vjeux - that makes total sense, agreed

**Comment by brentvatne** [comment]
@vjeux - repo is here: https://github.com/brentvatne/react-native-video

**Pull request closed without merging**